sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -57,7 +63,7 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
     try {
       if (!tryAcquire(numPermits, priority)) {
         val condition = lock.newCondition()
-        val info = ThreadInfo(priority, condition, numPermits)
+        val info = ThreadInfo(priority, condition, numPermits, TaskContext.get().taskAttemptId())
```

Comment:
Nit: The callers of this are already looking up the task attempt ID before we get here, so it's a bit wasteful to redundantly lookup the thread-local task context again here.  We could refactor this to have the task attempt ID passed in to tryAcquire/acquire and have the task attempt ID tracked in GpuSemaphore.TaskInfo which requires the task attempt ID to lookup the corresponding TaskInfo instance.

Reply:
Noted. I'll pay attention to such cases in the future. For now since I'm in a hurry to get it checked in ( the CI takes so long), I'm resolving this with no-fix

Reply:
fixed